### PR TITLE
Add master/detail tests

### PR DESCRIPTION
**Pull request by edgarmueller**

### CODE REVIEW

File: packages/angular/src/control.ts [hunk 1]
```diff
@@ -55,7 +56,9 @@ export class JsonFormsControl extends JsonFormsBaseRenderer implements OnInit, O
     ngOnInit() {
         this.subscription = this.ngRedux
             .select()
-            .map((s: JsonFormsState) => this.mapToProps(s))
+            .pipe<ControlProps>(
```
Comment (eneufeld) [inline packages/angular/src/control.ts]: what do we gain using the pipe?

File: packages/ionic/test/horizontal-layout.spec.ts [hunk 2]
```diff
@@ -62,17 +62,15 @@ describe('Horizontal layout', () => {
             ],
             providers: [
                 {provide: Platform, useClass: PlatformMock},
-                {provide: NgRedux, useFactory: MockNgRedux.getInstance}
+                {provide: NgRedux, useFactory: MockNgRedux.getInstance},
+                {provide: Nav, useFactory: mockNav }
```
Comment (eneufeld) [inline packages/ionic/test/horizontal-layout.spec.ts]: do we need this in the horizontal test?

File: packages/ionic/test/master-detail.spec.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                {provide: NgRedux, useFactory: MockNgRedux.getInstance},
+                {provide: Nav, useValue: mockNav()},
+                MasterDetailNavService,
+            ],
+        }).compileComponents();
+
+        MockNgRedux.reset();
+    }));
+
+    beforeEach(() => {
```
Comment (eneufeld) [inline packages/ionic/test/master-detail.spec.ts]: I think we can merge the two beforeeach, similar to the horizontal layout test

File: packages/angular/src/control.ts (hunk 1 above)
Comment (edgarmueller) [reply packages/angular/src/control.ts]: Avoiding 'unused imports' warnings

**Pull request merged**